### ISSUE TITLE
[FEATURE] Allow definining multiple selectors of the same type

### ISSUE THREAD

**Issue by Aeltumn** [Ayfri/Kore#81]
### What module does this feature apply to?

DSL

### Feature description

I looked through the code and couldn't find this as a feature, but I'm trying to replicate a command in a datapack that goes:
```
/execute as @a[tag=!foo,tag=!bar,tag=!baz] run say test
```
But it appears that `SelectorNbtData` only lets you define one requirement per type. Vanilla simply parses all predicates in a row and adds them to a shared big predicate, so it allows repeating types. It does store whether a specific type appears at all for optimization reasons, but it lets you repeat them too.

### My issue is not duplicated

- [X] This is not a duplicate

**Comment by Ayfri** [comment]
Yes, that's a limitation I made way back when I started Kore, but now I know I should remove it, but that will be a breaking change without no doubt. I could maybe one day experiment with the [assignment operator](https://plugins.gradle.org/plugin/org.jetbrains.kotlin.plugin.assignment) but that is a way harder and complex way than changing all the fields to be lists.
I'll investigate what syntax should the user use, but it will be released in a second version of 1.20.5-pre1 (the first will only contain the important bug fixes).
